packages/server/src/util.js

```diff
@@ -18,7 +18,7 @@ export const clearModuleCache = moduleName => {
 }
 
 export const smartRequire = modulePath => {
-  if (process.env.NODE_ENV !== 'production') {
+  if (process.env.NODE_ENV !== 'production' || process.env.LOADABLE_KEEP_CACHE !== '1') {
```

Comment:
What about enabling clean cache only when HRM is enabled?
```suggestion
  if (process.env.NODE_ENV !== 'production' && module.hot) {
```